### PR TITLE
docs: clarify coding handoff to review

### PR DESCRIPTION
**Pull request by lsm**
Clarifies Coding workflow prompts so coders must hand off ready PRs with `send_message(target="Review", data: { pr_url })`, including after review-fix cycles. Adds tests pinning the prompt text and future session prompt injection.

Verified:
- `cd packages/daemon && bun test tests/unit/5-space/workflow/built-in-workflows.test.ts`
- `cd packages/daemon && bun test tests/unit/5-space/agent/custom-agent.test.ts`
- `bun run check`

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

### Summary

Changes correctly update all Coding node prompts across CODING_WORKFLOW and FULLSTACK_QA_LOOP_WORKFLOW to require explicit `send_message(target="Review", data: { pr_url })` for gated handoff. Gate names are accurate (`code-ready-gate` vs `code-pr-gate`). Tests pin the prompt text and verify injection through `createCustomAgentInit`. Clean, surgical change.

### Findings

**P2 — Research workflow has the same gap (follow-up)**
`built-in-workflows.ts:843-855` — The Research workflow agent prompt stops at "open a PR with `gh pr create`" but never tells the agent to call `send_message` with `data: { pr_url }`. The `research-ready-gate` at line 900 requires `pr_url`, same deadlock pattern this PR fixes for Coding workflows. Out of scope for this task (acceptance criteria scope is Coding nodes), but worth a follow-up task to apply the same fix.

### Verified

- Gate names correct: `code-ready-gate` in CODING_WORKFLOW, `code-pr-gate` in FULLSTACK_QA_LOOP_WORKFLOW
- `FULLSTACK_CODING_PROMPT` constant (line 528) updated consistently
- CODING_WORKFLOW coder prompt steps 6 + re-trigger (lines 607-635) both updated
- FULLSTACK_QA_LOOP_WORKFLOW coder prompt step 4 (line 1252) updated
- Tests cover both workflows + `createCustomAgentInit` injection path

No P0-P1 findings. Ship-ready.

File: packages/daemon/src/lib/space/workflows/built-in-workflows.ts [hunk 1]
```diff
@@ -601,9 +604,11 @@ export const CODING_WORKFLOW: SpaceWorkflow = {
               '3. Write or update tests to cover new behavior\n' +
               '4. Run the test suite and fix any failures\n' +
               '5. If code changed: open a PR with `gh pr create` — include a clear title and description\n' +
-              '6. If code changed: hand off by sending a message to Review with ' +
-              '`data: { pr_url: "<url>" }`. The gate script verifies the PR is open and ' +
-              'mergeable, so make sure it actually is before sending. ' +
+              '6. If code changed: hand off by calling ' +
+              '`send_message(target="Review", message="<short summary>", data: { pr_url: "<url>" })`. ' +
+              'The `data.pr_url` payload is auto-merged into `code-ready-gate`; the gate script verifies ' +
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/workflows/built-in-workflows.ts]: Fixed in 89781b2a2. The Coding and Fullstack slot prompts no longer hard-code `send_message(target="Review")`, `code-ready-gate`, or `code-pr-gate`; they now give behavioral guidance and point to the injected Runtime Execution Contract for exact target/data fields. Exact `send_message(target=..., data=...)` guidance is generated centrally from the workflow graph in `buildCustomAgentTaskMessage` and `TaskAgentManager`, with tests covering both behavioral slot prompts and graph-derived task-message injection.

File: packages/daemon/src/lib/space/workflows/built-in-workflows.ts [hunk 2]
```diff
@@ -1241,7 +1247,9 @@ export const FULLSTACK_QA_LOOP_WORKFLOW: SpaceWorkflow = {
               '1. Implement backend and frontend changes with focused commits\n' +
               '2. Add/update unit, integration, and UI tests as needed\n' +
               '3. Open or update the PR and ensure it remains mergeable\n' +
-              '4. Write code-pr-gate with field pr_url so Review can activate\n' +
+              '4. Hand off to Review by calling ' +
+              '`send_message(target="Review", message="<short summary>", data: { pr_url: "<url>" })`; ' +
+              '`save_artifact` alone will not open `code-pr-gate`\n' +
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/workflows/built-in-workflows.ts]: Fixed in 89781b2a2. Restamp now has a narrow patch path for known retired built-in prompt markers, replacing those stale built-in prompt values with the current template prompt while preserving arbitrary custom prompts. Added coverage for the stale `hand off by sending a message to Review with data.pr_url` prompt updating during restamp, and kept the sentinel custom-prompt preservation test intact.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: REQUEST_CHANGES

### Summary

V2 approach is much better — dynamic generation from workflow graph covers ALL workflows (including Research) automatically. Slot prompts are behavioral, not hard-coded. Restamp patches stale prompts. However, one case-sensitivity bug in the restamp logic means the FULLSTACK_QA_LOOP_WORKFLOW stale prompt will not get patched on existing deployments.

### Findings

**P1 — Case mismatch in BUILT_IN_PROMPT_PATCH_MARKERS**
`built-in-workflows.ts:1686`

Marker: `'write code-pr-gate with field pr_url so Review can activate'` (lowercase w)
Old dev-branch prompt (FULLSTACK_QA_LOOP_WORKFLOW coder step 4): `'4. Write code-pr-gate with field pr_url so Review can activate'` (uppercase W)

`.includes()` is case-sensitive, so the patch will not fire for the FULLSTACK stale prompt. Existing workflows seeded before this PR will not get the prompt updated on restamp.

Fix: Add the uppercase variant as a fourth marker:
```ts
const BUILT_IN_PROMPT_PATCH_MARKERS = [
  'write code-pr-gate with field pr_url so Review can activate',
  'Write code-pr-gate with field pr_url so Review can activate',
  ...
] as const;
```

**P3 — Duplicated format functions**
`custom-agent.ts:555-578` and `task-agent-manager.ts:2435-2460`

`formatSendMessageTarget`, `formatGateDataShape`, `formatGateDataPlaceholder` are identical logic in both files. Acceptable for now but file a follow-up to extract to shared module.

### Verified

- Research workflow gap automatically covered by dynamic generation ✅
- CODING_WORKFLOW and FULLSTACK_QA_LOOP_WORKFLOW slot prompts use behavioral wording ✅
- All 267 workflow tests + 53 custom-agent tests pass ✅
- `buildGatedHandoffLines` correctly handles all gated channels from workflow graph ✅
- `patchKnownBuiltInPromptDrift` correctly patches CODING_WORKFLOW stale prompts ✅

Request: fix the case mismatch marker, then this is ready.

File: packages/daemon/src/lib/space/workflows/built-in-workflows.ts [hunk 3]
```diff
@@ -1654,6 +1682,25 @@ function migrateCodexFeatureToNodeToggle(
...
+  existingPrompt: T,
+  templatePrompt: T
+): T {
+  const existingValue = existingPrompt?.value;
+  const templateValue = templatePrompt?.value;
+  if (!existingValue || !templateValue || existingValue === templateValue) return existingPrompt;
+  if (!BUILT_IN_PROMPT_PATCH_MARKERS.some((marker) => existingValue.includes(marker))) {
+    return existingPrompt;
+  }
+  return { ...existingPrompt, value: templateValue } as T;
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/workflows/built-in-workflows.ts]: Fixed in 2540642e1. Restamp now only patches exact retired built-in prompt values generated from the current template and known retired replacements; substring matches no longer overwrite customized prompts. Added coverage for exact retired Coding/Fullstack prompts updating and a custom prompt that still contains retired text remaining unchanged.

File: packages/daemon/src/lib/space/runtime/task-agent-manager.ts [hunk 4]
```diff
@@ -2386,6 +2386,11 @@ export class TaskAgentManager {
           continue;
         }
 
+        const targetArg = this.formatSendMessageTarget(channel.to);
+        const dataShape = this.formatGateDataShape(writableFields);
+        lines.push(
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/task-agent-manager.ts]: Fixed in 2540642e1. Gated handoff examples now expand multicast channel targets into per-target scalar send_message calls, so injected guidance never recommends an array target that bypasses gate data merging. Shared formatting lives in gated-handoff-guidance.ts and custom-agent coverage asserts scalar examples for a multicast gated channel.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

### Summary

All previous P1/P3 findings addressed. Restamp now uses exact-value matching instead of case-sensitive substring markers. Format functions extracted to shared `gated-handoff-guidance.ts`. Multicast gated channels emit per-target scalar `send_message` calls. Customized prompts containing retired text are preserved (not overwritten). Full check suite passes.

### Verified

- **P1 case mismatch fixed**: Restamp uses exact retired prompt value comparison via `isExactRetiredBuiltInPrompt()` — no substring matching ✅
- **P3 duplication fixed**: `formatGatedHandoffCall`, `getSendMessageTargets`, `formatGateDataShape` extracted to `gated-handoff-guidance.ts` ✅
- **Research workflow covered**: Dynamic generation from workflow graph handles all gated channels ✅
- **Restamp tests**: Both CODING_WORKFLOW and FULLSTACK_QA_LOOP_WORKFLOW stale prompt patching tested ✅
- **Custom prompt preservation**: Test at line 2147 verifies user customizations containing retired text are NOT overwritten ✅
- **Multicast**: `getSendMessageTargets` splits array targets into individual per-target lines ✅
- **323 tests pass, `bun run check` clean** ✅

### Note: scope expansion

Latest commit includes unrelated changes (Codex bridge model ID `claude-opus-4-1-20250805` → `claude-opus-4-7`, session lifecycle testability hook, 1-core test updates). These appear to be a sync with dev or separate fix. Not blocking but worth noting — ideally these would be a separate PR.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+export interface GateDataShapeField {
+  name: string;
+  type: string;
+}
+
+export function getSendMessageTargets(target: string | string[]): string[] {
+  return Array.isArray(target) ? target : [target];
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in 745afd379. Broadcast gated handoff guidance now expands `to: "*"` into scalar node targets instead of emitting `send_message(target="*")`, so generated examples use gate-write-compatible targets. Added task-message coverage proving broadcast examples omit `target="*"`.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+export function formatGatedHandoffCall(
+  target: string,
+  fields: readonly GateDataShapeField[]
+): string {
+  return `send_message(target=${JSON.stringify(target)}, message="<short summary>", data: ${formatGateDataShape(fields)})`;
+}
+
+function formatGateDataShape(fields: readonly GateDataShapeField[]): string {
+  const entries = fields.map((field) => `${field.name}: ${formatGateDataPlaceholder(field)}`);
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in 745afd379. Generated gate data examples now serialize keys with `JSON.stringify(field.name)`, so custom fields like `pr-url` produce valid structured data examples. Added coverage for quoted non-identifier field names.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

### Round 4 verification

Latest commit addresses broadcast target expansion and field name quoting. Clean, focused change — 4 files, +100/-11.

- Broadcast `*` channels expanded to per-node scalar `send_message` calls via `getBroadcastTargets()` ✅
- Gate field names quoted with `JSON.stringify` in data shape examples — prevents invalid JS for hyphenated names ✅
- `getSendMessageTargets` deduplicates via `Set` ✅
- 325 tests pass, 2 new tests covering broadcast and quoted field names ✅
- No scope creep — all changes task-relevant ✅

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+function formatGateDataPlaceholder(field: GateDataShapeField): string {
+  switch (field.type) {
+    case 'string':
+      return `"<${field.name}>"`;
+    case 'number':
+      return '<number>';
+    case 'boolean':
+      return '<boolean>';
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in ce811423a. Non-string gate placeholders are now executable values (`true` for boolean, `0` for number, `null` fallback), while string/map examples remain valid. Added coverage for quoted keys plus boolean/number handoff data examples.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

Round 5: Gate data placeholders now emit executable values (`true`, `0`, `null`). String fields unchanged. 2 files, +21/-5. All 56 tests pass. Zero findings.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: REQUEST_CHANGES

### CI failure — scope creep tests broken

All 5 CI failures are in `session-lifecycle-sdk-title.test.ts` (new file added in commit `2540642e1`, unrelated to task #588):

- `lastTitleQueryOptions?.model` is `undefined` — the `titleGenerationQueryForTesting` hook isn't wiring the mock query correctly
- `isFallback` is `true` when `false` expected — generateTitleAndRenameBranch falls back because the SDK query mock never fires

**Root cause**: The `titleGenerationQueryForTesting` config field was added to `SessionLifecycleConfig` but the test mock doesn't propagate through the provider service correctly in CI (works locally due to different provider state).

**Two options**:
1. Fix the test mock to work in CI
2. Remove the unrelated scope-creep files (Codex bridge model IDs, session lifecycle hook, 1-core tests) — they belong in a separate PR

I'd strongly recommend option 2 — this PR should only contain the space subsystem changes. The Codex bridge and session lifecycle changes are unrelated and should be their own PR where they get focused review.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  name: string;
+  type: string;
+}
+
+export function getSendMessageTargets(
+  target: string | string[],
+  broadcastTargets: readonly string[] = []
+): string[] {
+  const targets = Array.isArray(target) ? target : [target];
+  return [...new Set(targets.flatMap((item) => (item === '*' ? broadcastTargets : [item])))];
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts:17]: Fixed in 032b6c4fd. The auto gate-write matcher now treats broadcast channel targets (`to: "*"`) as matching scalar `send_message` targets, so the generated broadcast handoff examples merge `data` into the gate before delivery. Added a regression test covering a `to: "*"` gated channel with scalar `target: "Review"` and asserting `gateWrite` plus persisted `pr_url` gate data.

File: packages/daemon/src/lib/space/tools/node-agent-tools.ts [hunk 9]
```diff
@@ -860,7 +860,11 @@ export function createNodeAgentToolHandlers(config: NodeAgentToolsConfig) {
               ...routedTargetNames.map(resolveNodeName),
             ]);
             return tos.some(
-              (to) => candidateTargets.includes(to) || to === myNodeName || to === myAgentName
+              (to) =>
+                to === '*' ||
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/tools/node-agent-tools.ts]: Fixed in d2faf35f5. Gate-write channel selection now first searches gated channels from the current node for concrete target matches, then falls back to broadcast (`to: "*"`) only when no exact/multicast target channel matches. Added regression coverage with a broadcast channel listed before a Review-specific channel; `send_message(target: "Review", data: ...)` now writes the Review gate and leaves the broadcast gate untouched.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  const entries = fields.map(
+    (field) => `${JSON.stringify(field.name)}: ${formatGateDataPlaceholder(field)}`
+  );
+  return `{ ${entries.join(', ')} }`;
+}
+
+function formatGateDataPlaceholder(field: GateDataShapeField): string {
+  switch (field.type) {
+    case 'string':
+      return `"<${field.name}>"`;
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in bf73db671. String placeholders now use JSON.stringify on the full placeholder value, so field names containing quotes or other delimiters generate copyable handoff data like `"pr\\"url": "<pr\\"url>"`.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    case 'number':
+      return '0';
+    case 'boolean':
+      return 'true';
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in bf73db671. Handoff examples now use equality check values for scalar fields and count check match/min values for map fields, so examples are activation-ready for cases like `score == 5`, `approved == false`, and counted vote maps.

File: packages/daemon/src/lib/space/tools/node-agent-tools.ts [hunk 12]
```diff
@@ -851,18 +851,22 @@ export function createNodeAgentToolHandlers(config: NodeAgentToolsConfig) {
...
+            ...routedTargetNames,
+            ...routedTargetNames.map(resolveNodeName),
+          ]);
+          const gatedChannel =
+            channelsFromCurrentNode.find((ch) => {
+              const tos = Array.isArray(ch.to) ? ch.to : [ch.to];
+              return tos.some(
+                (to) => candidateTargets.includes(to) || to === myNodeName || to === myAgentName
+              );
+            }) ?? channelsFromCurrentNode.find((ch) => ch.to === '*');
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/tools/node-agent-tools.ts]: Fixed in bf73db671. Broadcast gate fallback now only applies when the scalar target resolves to a real broadcast recipient (node or agent slot outside the sender node). Unknown/typo targets no longer write broadcast gate data before delivery fails. Added regression coverage for the unknown-target case.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+function formatGateDataPlaceholder(field: GateDataShapeField): string {
+  if (field.check?.op === '==') return formatLiteral(field.check.value);
+
+  switch (field.type) {
+    case 'string':
+      return JSON.stringify(`<${field.name}>`);
+    case 'number':
+      return '0';
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed. Numeric `!=` placeholders now choose a value that differs from the check value: `score != 0` emits `1`, while other `!=` numeric checks keep `0` when it satisfies the gate. Added coverage for `scoreNonZero` with `check: { op: "!=", value: 0 }`.

File: packages/daemon/src/lib/space/tools/node-agent-tools.ts [hunk 14]
```diff
@@ -851,18 +851,37 @@ export function createNodeAgentToolHandlers(config: NodeAgentToolsConfig) {
...
-            );
+            return true;
           });
+          const candidateTargets = uniqueTargetRefs([
+            ...routedTargetNames,
+            ...routedTargetNames.map(resolveNodeName),
+          ]);
+          const broadcastTargetRefs = new Set<string>();
+          for (const targetNode of workflow.nodes) {
+            if (targetNode.id === workflowNodeId) continue;
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/tools/node-agent-tools.ts]: Fixed. Broadcast gate fallback now includes same-node peer agent slots while still excluding the sender agent itself. Added regression coverage for a gated `to: "*"` channel where `send_message(target: "peer", data: ...)` writes and opens the broadcast gate before delivery.

File: packages/daemon/src/lib/space/runtime/task-agent-manager.ts [hunk 15]
```diff
@@ -2427,6 +2437,10 @@ export class TaskAgentManager {
     return check.op;
   }
 
+  private getBroadcastTargets(workflow: SpaceWorkflow, currentNode: WorkflowNode): string[] {
+    return workflow.nodes.filter((node) => node.id !== currentNode.id).map((node) => node.name);
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/task-agent-manager.ts]: Fixed in 8887d9df7. Broadcast handoff guidance now expands to workflow agent slots as well as other nodes, excludes the current sender, and covers same-node peer targets in the task prompt test.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+function formatMapPlaceholder(field: GateDataShapeField): string {
+  if (field.check?.op !== 'count') return '{ "<key>": "<value>" }';
+
+  const matchValue = formatLiteral(field.check.match);
+  const count = Math.max(1, field.check.min ?? 1);
+  const entries = Array.from(
+    { length: count },
+    (_, index) => `${JSON.stringify(`<key${index + 1}>`)}: ${matchValue}`
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in 8887d9df7. Map count placeholders now cap at three concrete entries and append a summarized "add N more matching entries" marker so large min values do not expand unbounded prompt text; added regression coverage for min: 25.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+  );
+  return `{ ${entries.join(', ')} }`;
+}
+
+function formatGateDataPlaceholder(field: GateDataShapeField): string {
+  if (field.check?.op === '==') return formatLiteral(field.check.value);
+
+  switch (field.type) {
+    case 'string':
+      return JSON.stringify(`<${field.name}>`);
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in ddce35cb5. String != placeholders now detect when the default placeholder would equal the forbidden value and emit a distinct "<field-different>" value; regression coverage verifies the generated handoff does not include the equal value.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+
+function formatGateDataShape(fields: readonly GateDataShapeField[]): string {
+  const entries = fields.map(
+    (field) => `${JSON.stringify(field.name)}: ${formatGateDataPlaceholder(field)}`
+  );
+  return `{ ${entries.join(', ')} }`;
+}
+
+function formatGateDataPlaceholder(field: GateDataShapeField): string {
+  if (field.check?.op === '==') return formatLiteral(field.check.value);
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts:37]: Fixed in ddce35cb5. Equality checks with no value key now omit that data field from the generated handoff instead of rendering null, matching the strict undefined semantics; regression coverage verifies the field is omitted.

File: packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+
+function formatMapPlaceholder(field: GateDataShapeField): string {
+  if (field.check?.op !== 'count') return '{ "<key>": "<value>" }';
+
+  const matchValue = formatLiteral(field.check.match);
+  const count = Math.max(1, field.check.min ?? 1);
+  const shownCount = Math.min(count, MAX_MAP_PLACEHOLDER_ENTRIES);
+  const entries = Array.from(
+    { length: shownCount },
+    (_, index) => `${JSON.stringify(`<key${index + 1}>`)}: ${matchValue}`
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/gated-handoff-guidance.ts]: Fixed in fcaaf8225. Map-count handoff examples now emit one writer-owned entry (`"<your-key>": match`) instead of enough entries to satisfy the gate threshold, so a single reviewer is not instructed to open the aggregate approval gate alone. Updated regression coverage for both small and large `min` values.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

### Summary

All prior findings addressed, CI green (32/32 checks pass). Smart gate placeholders, same-node broadcast peer resolution, broadcast channel gate-write in `node-agent-tools.ts`, session lifecycle test isolation fix. 1100+ lines of new tests covering all edge cases.

### Findings

**P2 — Unprotected `resolveNodeAgents` in broadcast target construction**
`node-agent-tools.ts:866`

`resolveNodeAgents(targetNode)` called without try-catch in the `broadcastTargetRefs` loop. Same call at line 480 wraps in try-catch with comment "skip this node." If a workflow has a node with an empty agents array and no legacy `agentId`, this throws and crashes `send_message`.

Low practical impact — all built-in workflows have agents on every node. But defensive consistency matters since the same function is wrapped 4 lines above.

```ts
// Current (line 866):
for (const agent of resolveNodeAgents(targetNode)) {
// Fix:
try {
  for (const agent of resolveNodeAgents(targetNode)) {
    ...
  }
} catch { /* skip malformed node */ }
```

### Verified

- Smart placeholders: `==` with value → literal, `!=` → different value, `count` map → correct entries, `==` without value → omitted ✅
- Same-node broadcast: peer agent slots included, self excluded ✅
- `node-agent-tools.ts`: broadcast gate-write matches, scalar preferred over broadcast, unknown targets skipped ✅
- Session lifecycle: inline type guard replaces shared import, test mock isolation fixed ✅
- CI: 32/32 checks pass ✅
- `getSendMessageTargets` deduplication via `Set` ✅
- `getBroadcastTargets` uses `node.agents ?? []` (safe, no `resolveNodeAgents`) ✅

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** claude-sonnet-4-20250514 | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

Round 7: Both P2s fixed. `resolveNodeAgents` wrapped in try-catch. Map-count placeholders simplified to single `{ "<your-key>": <matchValue> }`. 204 tests pass, CI green (32/32). Zero findings. Ship it.
